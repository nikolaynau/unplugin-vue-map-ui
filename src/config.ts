export const libraryName = 'vue-map-ui';

export const allComponents = [
  'VMapAttributionControl',
  'VMapControlPosition',
  'VMapLayersControl',
  'VMapScaleControl',
  'VMapZoomControl',
  'VMapDefaultIcon',
  'VMapDivIcon',
  'VMapDivMarker',
  'VMapIcon',
  'VMapIconMarker',
  'VMapMarker',
  'VMapPinIcon',
  'VMapPinMarker',
  'VMapTileLayer',
  'VMap',
  'VMapPane',
  'VMapArcGisAeroTileLayer',
  'VMapArcGisTileLayer',
  'VMapGoogleTileLayer',
  'VMapMapboxTileLayer',
  'VMapOpenTransportTileLayer',
  'VMapOsmTileLayer'
];
