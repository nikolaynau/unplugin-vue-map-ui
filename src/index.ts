export * from './core/resolvers';
export * from './core/presets';
