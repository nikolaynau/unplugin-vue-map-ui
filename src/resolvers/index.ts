export * from './vue3-map-ui';
