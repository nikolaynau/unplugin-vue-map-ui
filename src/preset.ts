import type { TypeImport } from 'unplugin-vue-components';
import { allComponents, libraryName } from './config';

export const VueMapUiPreset: TypeImport = {
  from: libraryName,
  names: allComponents
};
