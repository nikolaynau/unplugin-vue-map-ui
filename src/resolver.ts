import type { ComponentResolver } from 'unplugin-vue-components';
import { libraryName } from './config';

export function VueMapUiResolver(): ComponentResolver {
  return {
    type: 'component',
    resolve: name => {
      if (name.match(/^VMap[A-Z]?/)) {
        return {
          name,
          from: libraryName
        };
      }
    }
  };
}
