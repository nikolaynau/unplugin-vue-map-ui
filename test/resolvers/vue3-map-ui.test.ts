import { afterEach, describe, expect, it } from 'vitest';
import { build } from 'vite';
import path from 'path';
import { fileURLToPath } from 'url';
import vue from '@vitejs/plugin-vue';
import { deleteSync } from 'del';
import Components from 'unplugin-vue-components/vite';
import { Vue3MapUiResolver } from '../../src';

const __dirname = fileURLToPath(new URL('.', import.meta.url));
const appDir = path.resolve(__dirname, '../fixtures/app');
const distDir = path.resolve(appDir, './dist');

describe('Vue3 Map UI Resolver', () => {
  afterEach(() => {
    deleteSync(distDir);
  });

  it('components should be auto imported', async () => {
    const result: any = await build({
      root: appDir,
      plugins: [
        vue(),
        Components({
          resolvers: [Vue3MapUiResolver()]
        })
      ],
      build: {
        minify: false,
        rollupOptions: {
          external: ['vue', 'vue3-map-ui']
        }
      }
    });

    expect(result.output[0].code).toContain(
      'import { VMapOsmTileLayer, VMap } from "vue3-map-ui"'
    );
    expect(result.output[0].code).toContain(
      'const _component_VMapOsmTileLayer = VMapOsmTileLayer'
    );
    expect(result.output[0].code).toContain('const _component_VMap = VMap');
  });
});
