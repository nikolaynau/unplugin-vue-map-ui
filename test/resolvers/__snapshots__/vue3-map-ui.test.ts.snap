// Vitest Snapshot v1

exports[`Vue3 Map UI Resolver > components should be auto imported 1`] = `
"import { openBlock, createBlock, withCtx, createVNode, createApp } from \\"vue\\";
import { VMapOsmTileLayer, VMap } from \\"vue3-map-ui\\";
(function polyfill() {
  const relList = document.createElement(\\"link\\").relList;
  if (relList && relList.supports && relList.supports(\\"modulepreload\\")) {
    return;
  }
  for (const link of document.querySelectorAll('link[rel=\\"modulepreload\\"]')) {
    processPreload(link);
  }
  new MutationObserver((mutations) => {
    for (const mutation of mutations) {
      if (mutation.type !== \\"childList\\") {
        continue;
      }
      for (const node of mutation.addedNodes) {
        if (node.tagName === \\"LINK\\" && node.rel === \\"modulepreload\\")
          processPreload(node);
      }
    }
  }).observe(document, { childList: true, subtree: true });
  function getFetchOpts(script) {
    const fetchOpts = {};
    if (script.integrity)
      fetchOpts.integrity = script.integrity;
    if (script.referrerpolicy)
      fetchOpts.referrerPolicy = script.referrerpolicy;
    if (script.crossorigin === \\"use-credentials\\")
      fetchOpts.credentials = \\"include\\";
    else if (script.crossorigin === \\"anonymous\\")
      fetchOpts.credentials = \\"omit\\";
    else
      fetchOpts.credentials = \\"same-origin\\";
    return fetchOpts;
  }
  function processPreload(link) {
    if (link.ep)
      return;
    link.ep = true;
    const fetchOpts = getFetchOpts(link);
    fetch(link.href, fetchOpts);
  }
})();
const _export_sfc = (sfc, props) => {
  const target = sfc.__vccOpts || sfc;
  for (const [key, val] of props) {
    target[key] = val;
  }
  return target;
};
const _sfc_main = {};
function _sfc_render(_ctx, _cache) {
  const _component_VMapOsmTileLayer = VMapOsmTileLayer;
  const _component_VMap = VMap;
  return openBlock(), createBlock(_component_VMap, null, {
    default: withCtx(() => [
      createVNode(_component_VMapOsmTileLayer)
    ]),
    _: 1
    /* STABLE */
  });
}
const App = /* @__PURE__ */ _export_sfc(_sfc_main, [[\\"render\\", _sfc_render], [\\"__file\\", \\"C:/work/projects/pantry/js/re/unplugin-vue3-map-ui/test/fixtures/app/src/App.vue\\"]]);
const app = createApp(App);
app.mount(\\"#app\\");
"
`;
